pkg/zson/docs/spec.md

```diff
@@ -201,24 +197,24 @@ by a value encoding.  Here is a pseudo-grammar for value encodings:
         | <list> <elem> ;
 <terminal> := <char>*
 <char> := [^\n\\;[]] | <esc-sequence>
```

Comment:
I got a little confused here, but maybe the target audience is accustomed to seeing this in pseudo-grammars. Is the reader basically supposed to interpret `[^\n\\;[]]` as if it were a regexp, i.e. is the caret a placeholder for start-of-expression and not a literal caret? It also seems like the way this is defined doesn't make room for regular old characters like `A-Z` etc. though obviously the text below implies as much. Maybe I'm just dense.

Reply:
`[^abc]` is a negated character class and means "match any character that doesn't match `[abc]`". The target audience will be familiar.

Reply:
If we're being pedantic, in most regexp forms the right bracket has to be the first character after the ^, though that diminishes readability significantly.

Reply:
@aswan or @nwt will fix this one.

Reply:
@aswan's right; `]` should follow `^` immediately. I'd make this
```suggestion
<char> := [^][;\n\\] | <esc-sequence>
```